app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -309,6 +315,16 @@ private fun ConversationScreen(
     val conversationScreenState = rememberConversationScreenState()
     val messageComposerInnerState = rememberMessageComposerInnerState()
     val context = LocalContext.current
+
+    LaunchedEffect(conversationMessagesViewModel.savedStateHandle) {
```

Comment:
Out of curiosity , how is this LaunchedEffect working exactly ? Why is savedStateHandler used a key ?

Reply:
On the `MediaGalleryViewModel` we are adding a key+value to the `previousBackStackPassedArgs` when selecting different options like reply, message details or reacting to the image asset. This basically stores in the navigation manager `savedStateHandle`, information about pending actions that need to be checked by its corresponding Composable when navigating back.
This way, we can observe any change on the `savedStateHandle` and decide what to show on the `ConversationScreen`.